### ISSUE TITLE
`intersects('*', '^1.2.3')` throws a TypeError

### ISSUE THREAD

**Issue by jinwoo** [npm/node-semver#225]
Per the document, `'*'` is a valid version range representation for *any* versions. But when it's used in the `intersects()` function, it throws a `TypeError`.

```shell
> semver.intersects('*', '^1.2.3');
TypeError: Invalid Version: [object Object]
    at new SemVer (/<project>/node_modules/semver/semver.js:279:11)
    at compare (/<project>/node_modules/semver/semver.js:566:10)
    at lt (/<project>/node_modules/semver/semver.js:600:10)
    at cmp (/<project>/node_modules/semver/semver.js:641:21)
    at Comparator.intersects (/<project>/node_modules/semver/semver.js:733:5)
    at /<project>/node_modules/semver/semver.js:843:33
    at Array.every (<anonymous>)
    at /<project>/node_modules/semver/semver.js:842:33
    at Array.some (<anonymous>)
```

**Comment by jinwoo** [comment]
This happens with either `'*'`, `'x'`, or `''`. I.e. all range representations for *ANY*.

**Comment by pacamara** [comment]
@jinwoo Hi! I've posted a fix for this at https://github.com/npm/node-semver/pull/238 :beers:

**Comment by Asthetic** [comment]
Added a fix as well, I was concerned the previous change may not support wildcards in the middle of a version.

https://github.com/npm/node-semver/pull/271

**Comment by isaacs** [comment]
Fixed in 6.1
